analyzers/src/SonarAnalyzer.CSharp/Rules/CastConcreteTypeToInterface.cs

```diff
@@ -49,13 +49,12 @@ protected override void Initialize(SonarAnalysisContext context)
                     var castExpression = (BinaryExpressionSyntax)c.Node;
                     var castedTo = c.SemanticModel.GetTypeInfo(castExpression.Right).Type;
                     var castedFrom = c.SemanticModel.GetTypeInfo(castExpression.Left).Type;
-                    CheckForIssue(castedTo, castedFrom, c);
+                    CheckForIssue(c, castedTo, castedFrom);
                 },
                 SyntaxKind.AsExpression);
         }
 
-        public static void CheckForIssue(ITypeSymbol castedTo, ITypeSymbol castedFrom,
-            SyntaxNodeAnalysisContext context)
+        public static void CheckForIssue(SonarSyntaxNodeAnalysisContext context, ITypeSymbol castedTo, ITypeSymbol castedFrom)
```

Comment:
clean as we code: the parameter order is counter intuitive (initially I thought that there is an bug when I've seen the call). `From` should be before `To`. You could change also this since it affects only the lines that are already modified.

Reply:
I cleaned the whole rule a bit more, it's a small one